File: java/src/main/java/io/webweavex/fetch/JavaNetTransport.java
package io.webweavex.fetch;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;
import java.util.*;
import java.util.concurrent.CompletableFuture;
public class JavaNetTransport implements HttpTransport {
    private final HttpClient client = HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(30)).followRedirects(HttpClient.Redirect.NORMAL).build();
    private static final String UA = "WebWeaveX/3.0.0";
    public Map<String, Object> fetchSync(String url) {
        try {
            HttpRequest req = HttpRequest.newBuilder().uri(URI.create(url)).header("User-Agent", UA).timeout(Duration.ofSeconds(30)).GET().build();
            HttpResponse<String> resp = client.send(req, HttpResponse.BodyHandlers.ofString());
            Map<String, Object> r = new LinkedHashMap<>();
            r.put("text", resp.body()); r.put("status", resp.statusCode());
            r.put("contentType", resp.headers().firstValue("content-type").orElse("text/plain"));
            r.put("ok", resp.statusCode() >= 200 && resp.statusCode() < 400); r.put("error", "");
            return r;
        } catch (Exception e) {
            Map<String, Object> r = new LinkedHashMap<>();
            r.put("text", ""); r.put("status", 0); r.put("contentType", "text/plain");
            r.put("ok", false); r.put("error", e.getMessage()); return r;
        }
    }
    public CompletableFuture<Map<String, Object>> fetchAsync(String url) {
        return CompletableFuture.supplyAsync(() -> fetchSync(url));
    }
}

File: kotlin/src/main/kotlin/io/webweavex/fetch/JavaNetTransport.kt
package io.webweavex.fetch

import java.net.URI
import java.net.http.HttpClient
import java.net.http.HttpRequest
import java.net.http.HttpResponse
import java.time.Duration

class JavaNetTransport : HttpTransport {
    private val client = HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(30)).followRedirects(HttpClient.Redirect.NORMAL).build()

    override suspend fun fetch(url: String): Map<String, Any> {
        return try {
            val req = HttpRequest.newBuilder().uri(URI.create(url)).header("User-Agent", "WebWeaveX/3.0.0").timeout(Duration.ofSeconds(30)).GET().build()
            val resp = client.send(req, HttpResponse.BodyHandlers.ofString())
            mapOf("text" to resp.body(), "status" to resp.statusCode(), "contentType" to (resp.headers().firstValue("content-type").orElse("text/plain")), "ok" to (resp.statusCode() in 200..399), "error" to "")
        } catch (e: Exception) {
            mapOf("text" to "", "status" to 0, "contentType" to "text/plain", "ok" to false, "error" to (e.message ?: "unknown"))
        }
    }
}
